run/shas.py

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/env python
+
+# Copyright 2017 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import logging
+import os
+import subprocess
+
+from datetime import date, timedelta
+
+"""
+shas.py defines methods for determining which SHA to run against on a given
+date.
+"""
+
+
+class SHAFinder(object):
+
+    def __init__(self,
+                 logger  # type: logging.Logger
+                 ):
+        self.logger = logger
+
+    def get_todays_sha(self, path):  # type: (str) -> str
+        """
+        Gets the first SHA for the current date of the git repo in the given
+        path.
+        """
+        today = date.today()
+        tomorrow = today + timedelta(days=1)
+        command = [
+            'git',
+            'log',
+            '--format=%H',
+            '--after="%s 00:00:00"' % today.isoformat(),
+            '--before="%s 00:00:00"' % tomorrow.isoformat(),
+            '--date-order',
+            '--reverse'
+        ]
+        abspath = os.path.abspath(path)
+        logger.debug('Using dir ' + abspath)
+        logger.debug('Executing ' + ' '.join(command))
+        output = subprocess.check_output(command, cwd=abspath)
+        shas = output.decode('UTF-8').strip().split(os.linesep)
+        if len(shas) > 0:
+            return shas[0]
+        return ''
+
+    def get_head_sha(self, path):  # type: (str) -> str
+        """
+        Gets the HEAD SHA for the git repo in the given path.
+        """
+        command = [
+            'git',
+            'rev-parse',
+            'HEAD',
+        ]
+        abspath = os.path.abspath(path)
+        logger.debug('Using dir ' + abspath)
+        logger.debug('Executing ' + ' '.join(command))
+        output = subprocess.check_output(command, cwd=abspath)
+        sha = output.decode('UTF-8').strip()
+        assert len(sha) == 40, 'Invalid SHA: "%s"' % sha
+        return sha
+
+
+if __name__ == '__main__':
```

Comment:
This is never called in prod or tests, right? A nice-to-have?

Reply:
Yeah, I wrote/used for local integration testing.